src/interfaces/token.spec.ts

```diff
@@ -0,0 +1,475 @@
+    });
+
+    it("should get the suggested Usdc exchange rate for list of tokens", async () => {
+      getPriceUsdcMock.mockResolvedValueOnce(0.000001).mockResolvedValueOnce(0.000002);
+
+      expect(await tokenInterface.priceUsdc(["0x000", "0x001"])).toEqual({
+        "0x000": 0.000001,
+        "0x001": 0.000002
+      });
+      expect(getPriceUsdcMock).toHaveBeenCalledTimes(2);
+      expect(getPriceUsdcMock).toHaveBeenNthCalledWith(1, "0x000", undefined);
+      expect(getPriceUsdcMock).toHaveBeenNthCalledWith(2, "0x001", undefined);
+    });
+  });
+
+  describe("balances", () => {
+    const vaultTokenWithBalance = balanceFactory.build({
+      address: "0x001"
+    });
+    const vaultTokenWithoutBalance = balanceFactory.build({
+      balance: "0"
+    });
+    const zapperTokenWithBalance = balanceFactory.build({
+      address: "0x002"
+    });
+    const zapperSameAddressTokenWithBalance = balanceFactory.build({
+      address: "0x001"
+    });
+
+    beforeEach(() => {
+      vaultsBalancesMock.mockResolvedValue([vaultTokenWithBalance, vaultTokenWithoutBalance]);
+    });
+
+    describe("when chainId is 1 or 1337", () => {
+      beforeEach(() => {
+        tokenInterface = new TokenInterface(mockedYearn, 1, new Context({}));
+      });
+
+      it("should fetch token balances from the TokenInterface.supported list", async () => {
+        zapperBalancesMock.mockResolvedValue([zapperTokenWithBalance, zapperSameAddressTokenWithBalance]);
+
+        expect(await tokenInterface.balances("0x000")).toEqual([zapperTokenWithBalance, vaultTokenWithBalance]);
+        expect(zapperBalancesMock).toHaveBeenCalledTimes(1);
+        expect(zapperBalancesMock).toHaveBeenCalledWith("0x000");
+        expect(ironBankBalancesMock).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("when chainId is 250", () => {
+      beforeEach(() => {
+        tokenInterface = new TokenInterface(mockedYearn, 250, new Context({}));
+      });
+
+      it("should fetch token balances from the TokenInterface.supported list", async () => {
+        const ironBankBalance = balanceFactory.build();
+        ironBankBalancesMock.mockResolvedValue([ironBankBalance]);
+
+        expect(await tokenInterface.balances("0x000")).toEqual([ironBankBalance, vaultTokenWithBalance]);
+        expect(ironBankBalancesMock).toHaveBeenCalledTimes(1);
+        expect(ironBankBalancesMock).toHaveBeenCalledWith("0x000");
+        expect(zapperBalancesMock).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("when chainId is not supported", () => {
+      beforeEach(() => {
+        tokenInterface = new TokenInterface(mockedYearn, 42 as any, new Context({}));
+      });
+
+      it("should throw an SdkError", async () => {
+        try {
+          await tokenInterface.balances("0x000");
+        } catch (error) {
+          expect(error).toStrictEqual(new SdkError(`the chain ${42} hasn't been implemented yet`));
```

Comment:
no need for string interpolation

Reply:
Refactored with https://github.com/yearn/yearn-sdk/pull/218/commits/163623be382c58891cc709ac70d8cac36f272bc8